src/wrapper/jnienv.rs

```diff
@@ -118,6 +118,29 @@ impl<'a> JNIEnv<'a> {
         Ok(class)
     }
 
+    /// Define a new java class using an AutoByteArray
+    pub fn define_class_autobytearray<S>(
+        &self,
+        name: S,
+        loader: JObject<'a>,
+        buf: AutoByteArray,
+        buf_length: jsize,
```

Comment:
I think this should be an `usize` for generality, and you cast it to `jsize` (`buf_length as jsize`) below.

This looks good to me in general. But, I'm not an official reviewer. You'll have to convince them, no me. :-)

Reply:
Really? jsize is the official parameter in the documentation

Reply:
Let's wait for the feedback of the reviewers.

Reply:
I think that using `jsize` is fine: as far as I can see, all `JNIEnv` methods use this type. Additionally, we are still limited by `jsize` in the call to `DefineClass` function.

Reply:
Having the separate `buf_length` argument seems to make this a potentially `unsafe` API compared to `define_class` because it would be easy to trigger a buffer overrun by telling `DefineClass` that the buffer is larger than it really is.